loader/vk_loader_platform.h

```diff
@@ -287,9 +287,15 @@ static inline char *loader_platform_executable_path(char *buffer, size_t size) {
 #if TARGET_OS_OSX
 #include <libproc.h>
 static inline char *loader_platform_executable_path(char *buffer, size_t size) {
+    // proc_pidpath takes a uint32_t for the buffer size
+    if (size > UINT32_MAX) {
+        return NULL;
+    }
```

Comment:
Added this error check so proc_pidpath never gets an invalid `buffersize` passed in